### DISCUSSION TITLE
Field array minimal length validation

### DISCUSSION THREAD

**Discussion by JesseVelden** [react-hook-form/react-hook-form#4475] [Q&A] [answered]
Hi there! I couldn't find any other issue regarding validating the minimal length of an array besides the question asked at issue https://github.com/react-hook-form/react-hook-form/issues/2369 and question https://github.com/react-hook-form/react-hook-form/discussions/2960. 

In there, it says something about the documentation, but looking for this documentation, I could not find anything regarding this, except these two links. Is there any updated information for V7, or plans to have an  `useFieldArray.append` API of something like `append("test", {shouldValidate: true}`? 

Otherwise the `useEffect` method also works fine, but I was not sure if this is still the recommended way. Thanks anyway for this awesome library!

**Comment by bluebill1049** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

Search: Field Array Min Length it may be helpful

**Reply by JesseVelden** [reply]
Awesome thanks! I was looking at the documentation only, and didn't notice there was an example of this.

**Reply by franciscohanna92** [reply]
@bluebill1049  Is there any other way to accomplish this without using Yup or any other external validator library?

**Reply by Axxxx0n** [reply]
Yea, I think there should be option to do it without external validatio library

**Reply by imnotteixeira** [reply]
Hi, just stumbled upon this while having the same problem. I am using yup as well, and having a `.min` rule on the schema object does not trigger errors (I want to have a minimum of 3 entries of `useFieldArray`'s fields, let's say)

Additionally, in the example you mentioned @bluebill1049 it does not enforce that type of constraint, it simply validates each field individually, and the .required on the array itself only allows for a min(0) rule (there must be at least 1 field).

Is there any way to achieve this or is it not possible at the moment?
